### PR TITLE
1st Commit ,Json Parser

### PR DESCRIPTION
**Pull request by tavisca-mdixit**
@amarpalsapure  You can review JsonParser code.

### CODE REVIEW

**Comment by tavisca-mdixit** [comment]
@amarpalsapure  Code updated according to guidelines ,,

File: Training1/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public class CurrencyConverter : ICurencyConverter
+    {
+        private string _jsonFile;
+        public string JsonFile
+        {
+            get
+            {
+                return _jsonFile;
+            }
+            set
```
Comment (apalsapure) [inline Training1/OperatorOverloading.DBL/CurrencyConverter.cs]: Shouldn't this be `private`? Steps to break your code
- Construct object 
- Set `JsonFile` to `null`

File: Training1/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        public double GetConversionRate(string from, string to)
+        {
+            //Checking for valid arguements
+            if (string.IsNullOrWhiteSpace(from) || string.IsNullOrWhiteSpace(to) || from.Length != 3 && to.Length != 3)
+            {
+                throw new System.ArgumentException(Messages.InvalidInput);
+            }
+
+            //Splitting the jsonFile           
+            string[] jsonFileArray = _jsonFile.Split('{', ',', '}');
```
Comment (apalsapure) [inline Training1/OperatorOverloading.DBL/CurrencyConverter.cs]: What's the use of Split here if both currencies are same.

File: Training1/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+            //If to and from currency are same
+            if (string.Equals(to, from, StringComparison.OrdinalIgnoreCase))
+            {
+                return 1;
+            }
+
+            //Calling the convert function to convert from USD 
+            if (string.Equals(from, "USD", StringComparison.OrdinalIgnoreCase))
```
Comment (apalsapure) [inline Training1/OperatorOverloading.DBL/CurrencyConverter.cs]: Why did you hard code USD here? What will happen if tomorrow this code is used to convert INR to YEN and I have licence to fetch conversion rate for all currencies.

File: Training1/OperatorOverloading.DBL/CurrencyParser.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OperatorOverloading.DBL
+{
+    class CurrencyParser
+    {
+        public static double ConversionRate(string[] jsonFileArray, string currency)
+        {
+            foreach (string currencies in jsonFileArray)
```
Comment (apalsapure) [inline Training1/OperatorOverloading.DBL/CurrencyParser.cs:48]: What will happen if `jsonFileArray` is null

File: Training1/OperatorOverloading.DBL/CurrencyConverter.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using OperatorOverloading.Interface;
+using System.IO;
+namespace OperatorOverloading.DBL
+{
+    public class CurrencyConverter : ICurencyConverter
+    {
+        private string _jsonFile;
+        public string JsonFile
```
Comment (apalsapure) [inline Training1/OperatorOverloading.DBL/CurrencyConverter.cs]: This property should be in Parser. All parsing should be done in the parser.

File: Training1/OperatorOverloading.Model/Money.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        public Money Convert(string to)
+        {
+            //Checking for null arguements and white spaces and the length of arguement. 
+            if (string.IsNullOrWhiteSpace(to) || string.IsNullOrEmpty(to) || to.Length != 3 || Regex.IsMatch(to, @"^[a-zA-Z]+$") == false)
+            {
+                throw new ArgumentException(Messages.InvalidInput);
+            }
+            FileFetch fetch = new FileFetch();
```
Comment (apalsapure) [inline Training1/OperatorOverloading.Model/Money.cs]: Why FileFetch is here?

File: Training1/OperatorOverloading.Model/Money.cs [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            CurrencyConverter currencyConverter = new CurrencyConverter(fetch.FileFectcher());
```
Comment (apalsapure) [inline Training1/OperatorOverloading.Model/Money.cs]: What can be inferred from this name `FileFectcher`? Why not `GetFile` of simply `Get`

**Comment by apalsapure** [comment]
Your review score is 1.
- Null checking is not done, that makes your score to 3
- Your code is not object oriented, this makes score 2
- No gitignore file, score is 1

**Comment by tavisca-mdixit** [comment]
@amarpalsapure  Made changes according to review ,, you can review again.

**Pull request merged**